ssz/sedes/list.py

```diff
@@ -9,112 +10,120 @@
+        return EMPTY_LIST_HASH_TRIES_ROOT
 
+
+empty_list = EmptyList()
+
+
+class List(HomogenousSequence):
+    def __init__(self,
+                 element_sedes: BaseSedes[Sequence[TElement], Tuple[TElement, ...]] = None,
+                 ) -> None:
         # This sedes object corresponds to each element of the iterable
         self.element_sedes = element_sedes
-        # This empty bool indicates whether this sedes is meant for empty lists
-        self.empty = empty
 
     #
     # Size
     #
-    @property
-    def is_static_sized(self):
-        return False
+    is_fixed_sized = False
 
-    def get_static_size(self):
+    def get_fixed_size(self):
         raise ValueError("List has no static size")
 
-    #
-    # Serialization
-    #
-    def serialize_content(self, value: Iterable[TSerializable]) -> bytes:
-        if isinstance(value, (bytes, bytearray, str)):
-            raise SerializationError("Can not serialize strings as lists")
-
-        if self.empty:
-            self._validate_emptiness(value)
-
-        return b"".join(
-            self.element_sedes.serialize(element) for element in value
-        )
-
-    @staticmethod
-    def _validate_emptiness(value: Iterable[TSerializable]) -> None:
-        try:
-            first(value)
-        except StopIteration:
-            pass
-        else:
-            raise SerializationError(
-                "Can only serialize empty Iterables"
-            )
-
     #
     # Deserialization
     #
     @to_tuple
-    def deserialize_content(self, content: bytes) -> Generator[TDeserialized, None, None]:
-        if self.empty and len(content) > 0:
-            raise DeserializationError(f"Serialized list is not empty")
-
-        element_start_index = 0
-        while element_start_index < len(content):
-            element, next_element_start_index = self.element_sedes.deserialize_segment(
-                content,
-                element_start_index,
-            )
+    def _deserialize_stream(self, stream: IO[bytes]) -> Iterable[TElement]:
+        if self.element_sedes.is_fixed_sized:
+            element_size = self.element_sedes.get_fixed_size()
+            data = stream.read()
+            if len(data) % element_size != 0:
+                raise DeserializationError("TODO: INVALID LENGTH")
+            for segment in partition(element_size, data):
+                yield self.element_sedes.deserialize(segment)
+        else:
+            try:
+                first_offset = s_decode_offset(stream)
+            except DeserializationError:
+                # Empty list
+                return
 
-            if next_element_start_index <= element_start_index:
-                raise Exception("Invariant: must always make progress")
-            element_start_index = next_element_start_index
+            num_remaining_offset_bytes = first_offset - stream.tell()
+            if num_remaining_offset_bytes % 4 != 0:
```

Comment:
As @jannikluhn mentioned: https://github.com/ethereum/py-ssz/pull/47/files#r272979440
Should we use `OFFSET_SIZE` instead of `4`?
```suggestion
            if num_remaining_offset_bytes % OFFSET_SIZE != 0:
```